frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx

```diff
@@ -318,104 +354,13 @@ export default class ExpressionEditorTextfield extends React.Component {
-              ))}
-              <a
-                className="link text-bold block my1"
-                target="_blank"
-                href={MetabaseSettings.docsUrl("users-guide/expressions")}
-              >
-                <Icon name="reference" size={12} className="mr1" />
-                {t`Learn more`}
-              </a>
-            </div>
-          </Popover>
-        )}
-        {suggestions.length ? (
-          <Popover
-            className="not-rounded border-dark"
-            hasArrow={false}
-            tetherOptions={{
-              attachment: "top left",
-              targetAttachment: "bottom left",
-            }}
-            sizeToFit
-          >
-            <ul className="pb1" style={{ minWidth: 150, overflowY: "auto" }}>
-              {suggestions.map((suggestion, i) =>
-                // insert section title. assumes they're sorted by type
-                [
-                  (i === 0 || suggestion.type !== suggestions[i - 1].type) && (
-                    <li className="mx2 h6 text-uppercase text-bold text-medium py1 pt2">
-                      {SUGGESTION_SECTION_NAMES[suggestion.type] ||
-                        suggestion.type}
-                    </li>
-                  ),
-                  <li
-                    ref={r => {
-                      if (i === this.state.highlightedSuggestion) {
-                        this._selectedRow = r;
-                      }
-                    }}
-                    style={{ paddingTop: 5, paddingBottom: 5 }}
-                    className={cx(
-                      "px2 cursor-pointer text-white-hover bg-brand-hover hover-parent hover--inherit",
-                      {
-                        "text-white bg-brand":
-                          i === this.state.highlightedSuggestion,
-                      },
-                    )}
-                    onMouseDownCapture={e => this.onSuggestionMouseDown(e, i)}
-                  >
-                    {suggestion.range ? (
-                      <span>
-                        {suggestion.name.slice(0, suggestion.range[0])}
-                        <span
-                          className={cx("text-brand text-bold hover-child", {
-                            "text-white bg-brand":
-                              i === this.state.highlightedSuggestion,
-                          })}
-                        >
-                          {suggestion.name.slice(
-                            suggestion.range[0],
-                            suggestion.range[1],
-                          )}
-                        </span>
-                        {suggestion.name.slice(suggestion.range[1])}
-                      </span>
-                    ) : (
-                      suggestion.name
-                    )}
-                  </li>,
-                ],
-              )}
-            </ul>
-          </Popover>
-        ) : null}
+        <Errors compileError={compileError} />
```

Comment:
It looks like this is new. Are these understandable errors to display?

Reply:
There are two expression widgets, both of which wrap `ExpressionEditorTextfield`. `ExpressionPopover` which is used in basic QB mode (?) and a different widget used in the notebook mode. Only `ExpressionPopover` had code for displaying any errors -- they weren't displayed in notebook mode -- so I moved the error display code into `ExpressionEditorTextfield` so they'd display in both places. Take a look at the screenshots posted in #13175 -- the error messages weren't showing up in both places originally

Reply:
👍 sweet makes sense